2016-05/may-05.md

```diff
@@ -0,0 +1,74 @@
+## May 5 ([discuss](https://github.com/reactjs/core-notes/pull/13))
+
+### Attendees
+
+* [Ben](https://twitter.com/soprano) (React)
+* [Dan](https://twitter.com/dan_abramov) (React)
+* [Jim](http://github.com/jimfb) (React)
+* [Paul](https://twitter.com/zpao) (React)
+* [Sebastian](https://twitter.com/sebmarkbage) (React)
+* [Shayne](https://github.com/shayne) (React Native)
+* [Tom](https://twitter.com/tomocchino) (React)
+
+### ReactPerf Rewrite
+
+* [Dan](https://twitter.com/dan_abramov) got his rewrite of `ReactPerf` [merged into React master](https://github.com/facebook/react/pull/6647).
+* It is mostly functionally identical to the old one but is written in a more maintainable way.
+* At Facebook, we don’t use the console `ReactPerf` API much so we’ll need community help testing this.
+* There is a corresponding [React Native PR](https://github.com/facebook/react-native/pull/7283) that depends on these changes.
+* The change removes `ReactPerf.measure()` wrappers and instead emits events (e.g. `onBeginLifeCycleTimer`).
+* Related PRs: [#6549](https://github.com/facebook/react/pull/6549), [#6612](https://github.com/facebook/react/pull/6612), [#6633](https://github.com/facebook/react/pull/6633), [#6046](https://github.com/facebook/react/pull/6046).
+* It will most likely go into `15.1.0-alpha` early next week.
+
+### TestUtils and Enzyme
+
+* Airbnb’s [Enzyme](https://github.com/airbnb/enzyme) is well maintained and supported.
+* Our [`TestUtils`](https://facebook.github.io/react/docs/test-utils.html) are not in a great shape, and most people prefer Enzyme.
+* Should Enzyme become the official `TestUtils`?
```

Comment:
I'd like to see the shallow rendering capability remain a part of core react - whether that's a custom renderer or not shouldn't make a lot of difference to a consumer.

Enzyme's main utility is in replacing the `find/scry` methods with ones which work consistently when rendering to  DOM, String or Shallow. Having that library be the only place where a shallow renderer exists would be disappointing.

If moving the shallow rendering code out of core is a goal, hopefully the community (myself included) will find a way to keep it decoupled from the test assertion / node traversal stuff.

I hadn't realised enzyme was already prominently featured on the TestUtils docs page. This is a bit disappointing to me, as although I have nothing against enzyme, it pretty much kills any hope similar libraries have of getting takeup when exploring that space.

Anyway, to summarise my thoughts on this:

* 👍 deprecate `(find|scry)*`
* 👎 moving shallow rendering out to enzyme
* 😑 moving shallow rendering out to userland

This is driven from the fact I have a shallow-rendering based testing lib which is a couple of months older than enzyme and has ~40% of the monthly npm download numbers enzyme does. I'd like to see even more competition / exploration in this space, and having core bless a particular lib could stifle that.